miden-lib/asm/miden/note.masm

```diff
@@ -122,10 +141,18 @@ end
 #!
 #! - sender is the sender of the note currently being processed.
 export.get_sender
-    push.0
-    # => [0]
+    # pad the stack
+    padw padw padw push.0.0.0
+    # => [PAD(15)]
 
-    syscall.get_note_sender
+    exec.kernel_proc_offsets::get_note_sender_offset
+    # => [offset, PAD(15)]
+
+    syscall.exec_kernel_proc
+    # => [sender, PAD(15)]
+
+    # clean the stack
+    swapdw dropw dropw swapw dropw movdn.3 drop drop drop
```

Comment:
Not for this PR, but I wonder if we should write a set of procedures (maybe even in `miden-stdlib`) that would do something like this. For example:

- `keep1` - drop everything from the stack except for the first element.
- `keep2` - drop everything from the stack except for the first two elements.
- etc.

And maybe something equivalent for padding the stack as well - e.g., `pad15` - add 15 zero elements after the first stack element; `pad14` - add 14 zero elements after the first two stack elements.

Reply:
I agree! That will simplify the padding of the procedures a lot. Probably it makes sense to implement them while working on the ABI issue (#685)